### PR TITLE
New: Handle create success for tasks

### PR DESCRIPTION
**Pull request by JustinHoldstock**

### CODE REVIEW

**Comment by boxcla** [comment]
Verified that @JustinHoldstock has signed the CLA. Thanks for the pull request!

**Comment by boxcla** [comment]
# Webpack Bundle Size Github Bot

| Bundle        | Diff | Percentage Diff | Before | After |
| ------------- |:----:|:---------------:|:------:|:-----:|
| Total | 0 B | 0.00% | 0 B | 0 B |

Compared against previously merged PR #332

**Powered by [webpack-bundle-size-github-bot](https://github.com/wenboyu2/webpack-bundle-size-github-bot/)**

File: src/components/ContentSidebar/ContentSidebar.js [hunk 1]
```diff
@@ -674,16 +674,42 @@ class ContentSidebar extends PureComponent<Props, State> {
         });
     };
 
+    /**
+     * Adds a task to the tasks state and increases total_count.
+     *
+     * @param {Task} task - The newly created task from the API
+     * @return {void}
+     */
+    onTaskCreateSuccess(task: Task): void {
```
Comment (DanDeMicco) [inline src/components/ContentSidebar/ContentSidebar.js]: Minor feedback regarding this file. In my PR I renamed some things to have more consistent naming to the other API functions. Something like `createTask` and `createTaskSuccessCallback`

File: src/components/ContentSidebar/ContentSidebar.js [hunk 2]
```diff
@@ -674,16 +674,42 @@ class ContentSidebar extends PureComponent<Props, State> {
...
+
     /**
      * Posts a new task to the API
      *
      * @private
      * @param {string} text - The task's text
      * @param {Array} assignees - Array of assignees
      * @param {string} dueAt - The comment's text
+     * @param {Function} successCallback - Called on successful task creation
+     * @param {Function} errorCallback - Called on failure to task comment
```
Comment (DanDeMicco) [inline src/components/ContentSidebar/ContentSidebar.js]: still references comment

File: src/components/ContentSidebar/__tests__/ContentSidebar-test.js [hunk 3]
```diff
@@ -295,4 +295,132 @@ describe('components/ContentSidebar/ContentSidebar', () => {
...
+        test('should not add to the task entries state if there is none', () => {
+            instance.setState({ tasks: undefined });
+            instance.setState = jest.fn();
+
+            instance.onTaskCreateSuccess({});
+
+            expect(instance.setState).not.toBeCalled();
+        });
+
+        test('should add the task to the task entries state', () => {
```
Comment (DanDeMicco) [inline src/components/ContentSidebar/__tests__/ContentSidebar-test.js]: I think this and the below test can be combined into 1 test. Just add the assertion and change the test name

File: src/components/ContentSidebar/ActivityFeed/activity-feed/ActivityFeed.js [hunk 4]
```diff
@@ -181,6 +181,28 @@ class ActivityFeed extends React.Component<Props, State> {
...
+
+    /**
+     * Callback for error while creating a Task.
+     *
+     * @param {Error} error - Error thrown while creating the Task
+     * @param {string} id - ID of the feed item to update as no longer pending
+     * @return {void}
+     */
+    createTaskErrorCallback(error: Error, id: string): void {
+        console.log('Implement createTaskErrorCallback() via updateFeedItemPendingStatus()', error, id);
```
Comment (priyajeet) [inline src/components/ContentSidebar/ActivityFeed/activity-feed/ActivityFeed.js]: console.error? Or call the default error handler that does logging

File: src/components/ContentSidebar/ContentSidebar.js (hunk 1 above)
Comment (JustinHoldstock) [reply src/components/ContentSidebar/ContentSidebar.js]: Done. What do you think we should do with the props with those names? IE) `onTaskCreate`, `onCommentCreate` etc.

File: src/components/ContentSidebar/ContentSidebar.js (hunk 2 above)
Comment (JustinHoldstock) [reply src/components/ContentSidebar/ContentSidebar.js]: done

File: src/components/ContentSidebar/__tests__/ContentSidebar-test.js (hunk 3 above)
Comment (JustinHoldstock) [reply src/components/ContentSidebar/__tests__/ContentSidebar-test.js]: done

File: src/components/ContentSidebar/ActivityFeed/activity-feed/ActivityFeed.js (hunk 4 above)
Comment (JustinHoldstock) [reply src/components/ContentSidebar/ActivityFeed/activity-feed/ActivityFeed.js]: using `console.error()` and updating for `createCommentErrorCallback`

**Comment by DanDeMicco** [comment]
Also, change PR name to include New

File: src/components/ContentSidebar/ContentSidebar.js [hunk 5]
```diff
@@ -674,16 +674,42 @@ class ContentSidebar extends PureComponent<Props, State> {
         });
     };
 
+    /**
+     * Adds a task to the tasks state and increases total_count.
+     *
+     * @param {Task} task - The newly created task from the API
+     * @return {void}
+     */
+    createTaskSuccess(task: Task): void {
```
[the remark this answers is not in the document]
Comment (JustinHoldstock) [reply src/components/ContentSidebar/ContentSidebar.js]: Done!

**Pull request merged**